compiler/rustc_mir_transform/src/elaborate_drops.rs

```diff
@@ -380,7 +380,7 @@ impl<'b, 'tcx> ElaborateDropsCtxt<'b, 'tcx> {
                 LookupResult::Parent(None) => {}
                 LookupResult::Parent(Some(_)) => {
                     if !replace {
-                        self.tcx.dcx().span_delayed_bug(
+                        self.tcx.dcx().span_bug(
```

Comment:
unsure, but I'd like to ICE here to get a test, given that it's unreachable, we could currently rely on mir building not emitting code like this somewhere else or sth. Am generally in favor of eagerly ICEing where possible :grin: probably not the best to review this :p

if you keep it, please change it to `assert!(replace)`

Reply:
I prefer leaving it as `span_bug`, it avoids having to put the span somewhere in the assert output, and is easier to change back if it gets hit.